blockchain/blockchain.go

```diff
@@ -164,7 +157,22 @@ func (b *Blockchain) verifyBlock(txn db.Transaction, block *core.Block,
 		}
 	}
 
-	return core.VerifyBlockHash(block, b.network)
+	if len(block.Transactions) != len(block.Receipts) {
+		return ErrIncompatibleBlock{"len of transaction do not match len of receipts"}
+	}
+
+	for i := 0; i < len(block.Transactions); i++ {
+		if err = core.VerifyTransactionAndReceipt(block.Transactions[i],
+			block.Receipts[i], b.network); err != nil {
+			return ErrIncompatibleBlock{err.Error()}
+		}
+	}
```

Comment:
dont these belong in `VerifyBlockHash`?

Reply:
What I meant was moving these checks in to `VerifyBlockHash`. similar to https://github.com/NethermindEth/juno/pull/556/commits/30813eda425bfbe6dc7203c27b7e802d75bd0cc9

since verifying transaction hashes are ultimately a step in verifying the block hash

Reply:
Yeah you are right, I am making changes now, just rebased

Reply:
Done